### PR TITLE
Wav2 vec2 phoneme ctc tokenizer optimisation

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?

This is my FIRST PR! 
The Wav2Vec2PhonemCTCTokenizer is slow when its argument `do_phonemize` is set to True. It re-initialises the backend at each forward pass. This is adressed using a class argument. 

There was also an H4 title in the documentation which had a link which did not render( `<h4></h4>` used to replace `####`)  

Tests were passed, no additional ones were created. Runtime experiments to phonemize the entire 'tr' (turkish) subset of the common voice dataset gives a x10 boost in performances. 

Models:
- Wav2Vec2PhonemeCTCTokenizer: @patrickvonplaten, @LysandreJik 

Documentation: @sgugger

### CODE REVIEW

File: src/transformers/models/wav2vec2_phoneme/tokenization_wav2vec2_phoneme.py
Comment (sgugger) [inline src/transformers/models/wav2vec2_phoneme/tokenization_wav2vec2_phoneme.py:176]:
```suggestion
        """
        Initializes the backend.
```

File: src/transformers/models/wav2vec2_phoneme/tokenization_wav2vec2_phoneme.py
Comment (sgugger) [inline src/transformers/models/wav2vec2_phoneme/tokenization_wav2vec2_phoneme.py:179]:
```suggestion
            phonemizer_lang (`str`): The language to be used.
```

File: CONTRIBUTING.md
Comment (sgugger) [inline CONTRIBUTING.md:371]: This doesn't need to be a header (there is no reason to make it one). Let's just put this in bold and keep Markdown syntax.

**Review by sgugger — APPROVED** [review]
Congrats on your first PR!
LGTM with a few nits, but let's wait for @patrickvonplaten approval before merging.

**Review by patrickvonplaten — APPROVED** [review]
Great! Thanks for fixing this :-)

File: CONTRIBUTING.md
Comment (patrickvonplaten) [reply CONTRIBUTING.md:371]: +1

**Pull request merged**
